addons/web/static/tests/widgets/company_switcher_tests.js

```diff
@@ -125,6 +127,33 @@ QUnit.module('widgets', {
             });
         });
 
+        QUnit.test("Toggle multiple companies in one go in multiple company mode", async function (assert) {
+            /**
+             *          [x] **Company 1**          [x] **Company 1**
+             *  toggle->[ ] Company 2     ====>    [x] Company 2
+             *  toggle->[x] Company 3              [x] Company 3
+             */
+            assert.expect(2);
+            SwitchCompanyMenu.prototype.TOGGLE_DELAY = 1000;
```

Comment:
use the original value you'll have stored in beforeEach (in a variable in this odoo module's scope) instead of hardcoding it